packages/providers/src/memory-cache.ts

```diff
@@ -0,0 +1,65 @@
+import type { BytesLike } from '@ethersproject/bytes';
+import { hexlify } from '@ethersproject/bytes';
+
+type Cache = {
+  [key: string]: {
+    expires: number;
+    value: BytesLike;
+  };
+};
+const cache: Cache = {}; // it's a cache hash ~~> cash?
+
+const DEFAULT_TTL_IN_MS = 30 * 1000; // 30seconds
+
+class MemoryCache {
+  ttl: number;
+  constructor(ttlInMs: number | boolean) {
+    if (typeof ttlInMs === 'boolean') {
```

Comment:
Shouldn't be `typeof ttlInMs === 'boolean' && ttlInMs`?

Do we really need the type overloading here?

Reply:
My experience with [type overloading](https://github.com/FuelLabs/fuels-ts/issues/748#issuecomment-1400789719) is that it makes things harder, not easier.

FWIW, I suggested a minimal alternative down [here](https://github.com/FuelLabs/fuels-ts/pull/930#discussion_r1183519727).